qdata/parser.py

```diff
@@ -69,6 +71,9 @@ def mainprogram(self, *args):
         return self._program
 
     def statement(self, *args):
+        """A catch-all for all OpenQASM statement; this includes register, operator, and gate
+        declarations, include statements, quantum operations to be applied, and barriers.
+        """
```

Comment:
For completeness, and to avoid having to always refer to a separate document, do we want to put the relevant grammar lines directly in the docstrings for all symbols?

Reply:
I don't know, I'm in two minds about it

* It is really really nice for maintenance and readability
* However, it could easily go out of sync with the grammar file.

I think the pros outweigh the cons though